rpc/v6/adapters.go

```diff
@@ -0,0 +1,314 @@
+package rpcv6
+
+import (
+	"errors"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/starknet"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/NethermindEth/juno/vm"
+)
+
+/****************************************************
+		VM Adapters
+*****************************************************/
+
+func AdaptVMTransactionTrace(trace *vm.TransactionTrace) TransactionTrace {
+	var validateInvocation *FunctionInvocation
+	if trace.ValidateInvocation != nil {
+		res := AdaptVMFunctionInvocation(trace.ValidateInvocation)
+		validateInvocation = &res
+	}
+
+	var executeInvocation *ExecuteInvocation
+	if trace.ExecuteInvocation != nil {
+		res := AdaptVMExecuteInvocation(trace.ExecuteInvocation)
+		executeInvocation = &res
+	}
+
+	var feeTransferInvocation *FunctionInvocation
+	if trace.FeeTransferInvocation != nil {
+		res := AdaptVMFunctionInvocation(trace.FeeTransferInvocation)
+		feeTransferInvocation = &res
+	}
+
+	var constructorInvocation *FunctionInvocation
+	if trace.ConstructorInvocation != nil {
+		res := AdaptVMFunctionInvocation(trace.ConstructorInvocation)
+		constructorInvocation = &res
+	}
+
+	var functionInvocation *FunctionInvocation
+	if trace.FunctionInvocation != nil {
+		res := AdaptVMFunctionInvocation(trace.FunctionInvocation)
+		functionInvocation = &res
+	}
+
+	stateDiff := AdaptVMStateDiff(trace.StateDiff)
+
+	return TransactionTrace{
+		Type:                  TransactionType(trace.Type),
+		ValidateInvocation:    validateInvocation,
+		ExecuteInvocation:     executeInvocation,
+		FeeTransferInvocation: feeTransferInvocation,
+		ConstructorInvocation: constructorInvocation,
+		FunctionInvocation:    functionInvocation,
+		StateDiff:             &stateDiff,
```

Comment:
Why don't use the `&` directly here
```suggestion

	return TransactionTrace{
		Type:                  TransactionType(trace.Type),
		ValidateInvocation:    &validateInvocation,
		ExecuteInvocation:     &executeInvocation,
		FeeTransferInvocation: &feeTransferInvocation,
		ConstructorInvocation: &constructorInvocation,
		FunctionInvocation:    &functionInvocation,
		StateDiff:             &stateDiff,
```

Reply:
Because it'd mean we have that for example:

```go
var validateInvocation FunctionInvocation
if trace.ValidateInvocation != nil {
	validateInvocation := AdaptVMFunctionInvocation(trace.ValidateInvocation)
}
```

And if `trace.ValidateInvocation` is nil, we'd still have created and assigned a zero-valued `FunctionInvocation` to `TransactionTrace.ValidateInvocation`